src/visit_vtk/lightweight/vtkVisItUtility.C

```diff
@@ -1650,3 +1655,151 @@ vtkVisItUtility::IntersectBox(double bounds[6], double origin[3],
     return true;
 }
 
+// ****************************************************************************
+//  Function: ApplyPropsForNonFamilyFonts
+//
+//  Purpose: Handles setting of things like bold, italic, etc if the currently
+//           selected font is not a VTK "family" font.
+//
+//           Everywhere VisIt handles setting of font or font attributes, it
+//           uses VTK's very old, font "family" mechanism. This allows for one
+//           of three font familys to be selected (Arial (0), Courier (1) or
+//           Times (2)). On many systems, the ultimate font VTK selects to 
+//           satisfy the requested font "family" does not support unicode
+//           characters. By default, VisIt uses Arial (0) font family.
+//
+//           To improve this situation, we decided to replace the default
+//           font "family" (Arial) with an Arial-like TrueType font that is
+//           visually highly similar to Arial and also supports unicode chars.
+//           The selected font we choose was Deja Vu Sans. Four .ttf font
+//           files for regular, bold, italic and bold-italic are installed in
+//           "resources" directory.
+//
+//           For "family" fonts, things like bold and italic are
+//           delegated to an associated vtkTextProperty object and VTK handles
+//           them. But, for non-family fonts, this doesn't work. We need to
+//           select the appropriate font file variant explicitly.
+//
+//           This function is intended to be placed after any calls to
+//           vtkTextProperty->SetBold(), SetItalic() and ensure that if the
+//           associated font is not a VTK family font, it will select the
+//           correct font file variant.
+//
+//  Programmer: Mark C. Miller, Thu Apr 10 15:59:57 PDT 2025
+// ****************************************************************************
+
+bool
+vtkVisItUtility::FontHasVariants(vtkTextProperty *tprop)
+{
+    if (tprop->GetFontFamily() == VTK_ARIAL) return true;
+    if (tprop->GetFontFamily() == VTK_COURIER) return true;
+    if (tprop->GetFontFamily() == VTK_TIMES) return true;
+    return false;
+}
+
+void
+vtkVisItUtility::AdjustPropsForNonFamilyFonts(vtkTextProperty *tprop)
+{
+    std::string fontFile = GetVisItResourcesDirectory(VISIT_RESOURCES_FONTS) +
+                           VISIT_SLASH_STRING;
+    
+    // Treat Arial as the "Regular" form of DejaVuSans
```

Comment:
Can you elaborate on what these comments mean? What is the regular/mono/serif form of DejaVu and why are they mapped to Arial, Courier, and Times?

Reply:
These are different styles (behaviors) of fonts. Arial is a sans-serif (e.g. no serifs) font. Times is a serif font. Neither Arial or Times are "monospaced" (meaning columnar in that every character takes up the same space) like Courier which is often desired for displaying lines of code. My strong belief is that Arial, Times and Courier were used in VTK (long ago and still in our interface today) as recognizeable synonyms for those key font attributes.

Reply:
You asked what "regular/mono/serif" are. Does that suggest you are not as familiar with that font-speak? If so, don't  fret. Most people don't. But, most people *do know* what Arial (or Helvetica), Times and Courier font faces are. I think that made sense for VTK way back when and I think it still probably makes sense in our GUI. Though, I could probably easily be convinced otherwise.

Reply:
> Does that suggest you are not as familiar with that font-speak?

That's right. Thanks for explaining! I don't have a strong opinion here.